internal/benchmark/metrics/execution/peers.go

```diff
@@ -111,36 +107,36 @@ func (p *PeerMetric) measure(ctx context.Context) {
 	}
 
 	if err = json.NewDecoder(res.Body).Decode(&resp); err != nil {
-		p.AddDataPoint(map[string]uint32{
-			PeerCountMeasurement: 0,
-		})
+		p.writeMetric(0)
 		logger.WriteError(metric.ExecutionGroup, p.Name, err)
 		return
 	}
 
 	peerCountHex := resp.Result
 	if peerCountHex == "" {
-		p.AddDataPoint(map[string]uint32{
-			PeerCountMeasurement: 0,
-		})
+		p.writeMetric(0)
 		logger.WriteError(metric.ExecutionGroup, p.Name, errors.New("peer count RPC response was empty. Most likely net_peerCount RPC method is not supported"))
 		return
 	}
 
 	peerCount, err := strconv.ParseInt(peerCountHex[2:], 16, 64)
 	if err != nil {
-		p.AddDataPoint(map[string]uint32{
-			PeerCountMeasurement: 0,
-		})
+		p.writeMetric(0)
 		logger.WriteError(metric.ExecutionGroup, p.Name, err)
 		return
 	}
 
+	p.writeMetric(peerCount)
+}
+
+func (p *PeerMetric) writeMetric(value int64) {
 	p.AddDataPoint(map[string]uint32{
```

Comment:
yes, it could potentially be a generic struct. We already have a DataPoint struct that looks like this:

```Go
DataPoint[T Metricable] struct {
		Timestamp time.Time
		Values    map[string]T
}
```
Instead of Values being a `map[string]T` it could be just `T` and the name of the metric(which is currently a key in the map) could be a separate field, like this:

```Go
DataPoint[T Metricable] struct {
		Timestamp time.Time
		Value     T
                Name      string
}
```
then the data point object would represent a single value instead of a collection, which would work just fine, but from what I can see at this point, it wouldn’t significantly improve quality, so I do not think it is worth refactoring